Microsoft.HealthVault/Connection/HealthVaultConnectionBase.cs

```diff
@@ -178,6 +178,11 @@ protected HealthVaultConnectionBase(IServiceLocator serviceLocator)
             this.SessionCredential = await sessionCredentialClient.GetSessionCredentialAsync(token).ConfigureAwait(false);
         }
 
+        Task IConnectionInternal.RefreshSessionCredentialAsync(CancellationToken token)
+        {
+            return this.RefreshSessionCredentialAsync(token);
```

Comment:
Yeah and the method above needs to be run from within the lock.

---
In reply to: [115805026](https://github.com/Microsoft/healthvault-dotnetstandard-sdk/pull/27#discussion_r115805026) [](ancestors = 115805026,115799063)